actionview/lib/action_view/template.rb

```diff
@@ -111,10 +111,13 @@ class Template
       end
     end
 
+    CACHE_CALL_PATTERN = /\A(?:<%#.*%>\n?)?<% cache\(?\s*(\w+\.?)/
```

Comment:
@jeremy I added a test for this.

@dmitry I pushed the pattern to a `resource_cache_call_pattern` on the ERB handler. HAML and SLIM can implement this themselves.